backend/core/interactem/core/models/agent.py

```diff
@@ -49,3 +55,9 @@ def clear_old_errors(self, max_age_seconds: float = 30.0) -> None:
         if not self.error_messages:
             if self.status == AgentStatus.ERROR:
                 self.status = AgentStatus.IDLE
+
+    @model_validator(mode="after")
+    def check_has_networks(self) -> "AgentVal":
+        if not self.networks:
+            raise ValueError("Agent must have at least one network.")
```

Comment:
Is this really true? If no network is provide it means it can be used to schedule any operator that doesn't have a network requirement?

Reply:
We could default instead to "public", I guess? But for now I prefer keeping it as a requirement and rethink this later when we are working on the messengers again.

Reply:
also operators don't have network requirements. this is a construct to see where operators are scheduled by the orchestrator,